spec/factories/users.rb

```diff
@@ -0,0 +1,33 @@
+FactoryBot.define do
+  factory :user do
+    email { Faker::Internet.safe_email }
+    password { Faker::Internet.password }
+    name { Faker::Name.first_name }
+    user_name { Faker::Internet.username }
+
+    factory :user_with_conversations do
+      transient do
+        conversations_count { 5 }
+      end
+
+      after(:create) do |user, evaluator|
+        # create(:room, user_id: user.id, conversation_id: create(:conversation).id)
+        create_list(:room, evaluator.conversations_count, user_id: user.id) do |room, index|
+        room.conversation_id = create(:conversation).id
+        end
+        user.reload
+      end
+    end
+
+    factory :user_with_messages do
+      transient do
+        messages_count { 5 }
+      end
+
+      after(:create) do |user, evaluator|
+        create_list(:message, evaluator.messages_count, user: user)
+        user.reload
```

Comment:
Do we need this reload?

Reply:
I think i read it from the everyday rails book